base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -258,21 +355,127 @@ where T: BlockchainBackend
     /// * The height is beyond the current chain tip.
     /// * The height is lower than the block at the pruning horizon.
     pub fn fetch_block(&self, height: u64) -> Result<HistoricalBlock, ChainStorageError> {
-        unimplemented!()
+        let metadata = self.check_for_valid_height(height)?;
+        let header = self.fetch_header(height)?;
+        println!("Got header");
+        let kernel_cp = self.fetch_mmr_checkpoint(MmrTree::Kernel, height)?;
+        println!("Got kernel checkpoint");
```

Comment:
There are some printlns in this function that can be removed.

Reply:
Thanks